services/motion/builtin/move_request.go

```diff
@@ -189,6 +217,75 @@ func (mr *moveRequest) deviatedFromPlan(ctx context.Context, waypoints state.Way
 	return state.ExecuteResponse{}, nil
 }
 
+// getTransientDetections returns a list of geometries as observed by the provided vision service and camera.
+// Depending on the caller, the geometries returned are either in their relative position
+// with respect to the base or in their absolute position with respect to the world.
+func (mr *moveRequest) getTransientDetections(
+	ctx context.Context,
+	visSrvc vision.Service,
+	camName resource.Name,
```

Comment:
@raybjork @biotinker 

To continue on the thread started here (https://github.com/viamrobotics/rdk/pull/3473#discussion_r1480260186)
> That we need both of these is a red flag for me. Not sure exactly what the solution is but I will look more closely at this and see what we can do.

In the current version of this PR, the `getTransientDetections` function returns geometries in either their relative or absolute positions. Here, I interpret relative to mean with respect to the base's frame (i.e. with respect to the coordinate system the base uses), and absolute to mean with respect to the world's frame and their own unique coordinate system.

The motivation for having detections returned in either the base or world frame is due to a divergence between how we digest obstacles at when constructing a plan versus when checking a plan for collisions with transient detections.
The former, constructing a plan, requires us to specify geometries in the base frame, while checking a plan for collisions requires we specify geometries in the world frame.

I think that a major simplification in terms of code and mental overhead for reasoning about transient obstacle collisions would be to change/refactor our current implementation of `CheckPlan` within the `motionplan` package. 
This would mean that `getTransientDetections` now returns geometries always with respect to the base frame. In turn, would could then remove `getCurrentPosition` and `getExistingGeometries` , which are added in this PR, from `move_request.go`

Reply:
I would prefer to keep this as getTransientDetections since we are returning gifs in the world frame and treating them as with respect to the base only within the motionplan stack.